tst/testutils.g

```diff
@@ -1,12 +1,173 @@
+#
+# 1. Consistency check: each group of degree n should act
+# transitively and primitively on [1..n]
+#
+PrimGrpConsistencyCheckDegree := function(deg) 
+    local i, g;
+    for i in [1..NrPrimitiveGroups(deg)] do
+        Print("Degree ", deg, " : ", i, "/", NrPrimitiveGroups(deg), "\r");
+        g := PrimitiveGroup(deg,i);
+        if MovedPoints(g) <> [1..deg] then
+            Print("PrimitiveGroup(",deg,",",i,") is not acting on ", [1..deg], "\n");
+        fi;
+        if not IsTransitive(g,[1..deg]) then
+            Print("PrimitiveGroup(",deg,",",i,") is not transitive on ", [1..deg], "\n");
+        fi;
+        if not IsPrimitive(g,[1..deg]) then
+            Print("PrimitiveGroup(",deg,",",i,") is not primitive on ", [1..deg], "\n");
+        fi;
+    od;
+    Print("                                                      \r");
+end;
+
+PrimGrpConsistencyCheck := function(i,j)
+local k;
+for k in [i..j] do
+    PrimGrpConsistencyCheckDegree(k);
+od;
+Print("\n");
+end;
+
+######################################################################
+#
+# 2. Checking names of primitive groups
+#
+
+takeCfromcyclicgroup := function( str )
+local pos;
+repeat
+  pos:=Position(str,'C');
+  if pos <> fail then
+    if IsDigitChar(str[pos+1]) then
+      str := Concatenation(str{[1..pos-1]},str{[pos+1..Length(str)]});
+    fi;
+  fi;
+until pos = fail;
+return str;
+end;
+
+comparenames := function(name,desc)
+local n, d;
+n := ReplacedString(name," ","");
+n := ReplacedString(n,"Alt","A");
+n := ReplacedString(n,"Sym","S");
+d := ReplacedString(desc," ","");
+if StartsWith(n,"A(") and EndsWith(n,")") and StartsWith(d,"A") or
+   StartsWith(n,"C(") and EndsWith(n,")") and StartsWith(d,"C") or
+   StartsWith(n,"M(") and EndsWith(n,")") and StartsWith(d,"M") or
+   StartsWith(n,"S(") and EndsWith(n,")") and StartsWith(d,"S") then
+  n := ReplacedString(n,"(","");
+  n := ReplacedString(n,")","");
+elif StartsWith(n,"M_") then
+  n := Concatenation( "M", n{[3..Length(n)]} );
+elif StartsWith(n,"D(") and EndsWith(n,")") and StartsWith(d,"D") then
+  n := Concatenation( "D", String(EvalString(n{[3..Length(n)-1]}) ) );
+fi;
+return n=d or n=takeCfromcyclicgroup(d);
+end;
+
+findaffinegroups := function(type,name)
+#
+# For a finite field F
+# AGL(d,F) = SemidirectProduct(GL(d, F), F^d)
+# ASL(d,F) = SemidirectProduct(SL(d, F), F^d)
+#
+local n, old, new, pos, pos2, d, f, arguments;
+n := ShallowCopy(name);
+old := Concatenation( type, "(");
+pos := PositionSublist( n, old );
+if pos = 1 then
+  pos2 := Position( n, ')' );
+  arguments := n{[ Length(old)+1 .. pos2-1]};
+  arguments := ReplacedString(arguments," ","");
+  arguments := SplitString(arguments,",");
+  new := Concatenation(
+            "SemidirectProduct(",
+            type{[2..3]}, # will be GL or SL
+            "(", arguments[1], ",",
+            "GF(", arguments[2], ")),",
+            "GF(", arguments[2], ")^", arguments[1], ")"
+            );
```

Comment:
I'd suggest to instead provide the functions AGL, ASL at the top level, and then not bother with transforming the code further